controllers/resource/dogu_secret_handler.go

```diff
@@ -0,0 +1,121 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	"github.com/cloudogu/cesapp/v4/core"
+	"github.com/cloudogu/cesapp/v4/keys"
+	"github.com/cloudogu/cesapp/v4/registry"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"strings"
+)
+
+type doguSecretWriter struct {
+	Client   client.Client
+	Registry registry.Registry
+}
+
+// NewDoguSecretsWriter creates a new instance of doguSecretWriter to save key value pairs from secrets to the dogu config
+func NewDoguSecretsWriter(client client.Client, registry registry.Registry) *doguSecretWriter {
+	return &doguSecretWriter{
+		Client:   client,
+		Registry: registry,
+	}
+}
+
+// WriteDoguSecretsToRegistry gets the dogu secret and writes the values in the dogu config. If no secret is found the
+// method returns no error
+func (drr *doguSecretWriter) WriteDoguSecretsToRegistry(ctx context.Context, doguResource *k8sv1.Dogu) error {
+	secret, err := drr.getDoguSetupSecret(ctx, doguResource)
+	if err != nil {
+		return err
+	}
+
+	if secret == nil {
+		return nil
+	}
+
+	err = drr.writeSecretsToRegistry(secret, doguResource.Name)
+	if err != nil {
+		return fmt.Errorf("failed to write secrets for dogu [%s]: %w", doguResource.Name, err)
+	}
+
+	err = drr.Client.Delete(ctx, secret)
+	if err != nil {
+		return fmt.Errorf("failed to delete %s setup secret: %w", doguResource.Name, err)
+	}
+
+	return nil
+}
+
+func (drr *doguSecretWriter) writeSecretsToRegistry(secret *corev1.Secret, dogu string) error {
+	doguPublicKey, err := GetPublicKey(drr.Registry, dogu)
+	if err != nil {
+		return err
+	}
+	doguConfig := drr.Registry.DoguConfig(dogu)
+
+	for etcdKey, secretValue := range secret.Data {
+		// setup writes keys from secret with "." instead of "/" because they are not allowed
+		key := strings.ReplaceAll(etcdKey, ".", "/")
+		encryptedValue, err := doguPublicKey.Encrypt(string(secretValue))
+		if err != nil {
+			return fmt.Errorf("failed to encrypt value for key %s: %w", key, err)
+		}
+		err = doguConfig.Set(key, encryptedValue)
+		if err != nil {
+			return fmt.Errorf("failed to write key %s to registry: %w", key, err)
+		}
```

Comment:
"Stay on Style" by adding a new line after doing an operation with following error check: 

```suggestion
		encryptedValue, err := doguPublicKey.Encrypt(string(secretValue))
		if err != nil {
			return fmt.Errorf("failed to encrypt value for key %s: %w", key, err)
		}
		
		err = doguConfig.Set(key, encryptedValue)
		if err != nil {
			return fmt.Errorf("failed to write key %s to registry: %w", key, err)
		}
```